make weekLetter and weekColumnHeader injectable

vasilakisfil opened the pull request:
Hello I added some more injectable options (check the Readme). We needed them for a Swedish ember app.

Let me know if you need something else to before merging in.

RSSchermer commented:
I like the idea of making the weekColumnHeader a setting :thumbsup:

I'm less sure about the weekLetter though. I think it should definitely be customizable, but perhaps more than just the weekletter. I chose the `{year}-W{weekNumber}` format as the default because that's the ISO standard notation, but maybe I'd want it formatted as `2015 week 14` or `vecka 15, 2015`.

One option that is technically already possible, is to simply extend the component and override the `weekText` method:

```js
// app/components/rl-week-picker.js
import RlWeekPickerComponent from 'ember-rl-week-picker/components/rl-week-picker';

export default RlWeekPickerComponent.extend({
  weekText: function () {
    var weekNumber = this.get('weekNumber');
    var year = this.get('year');

    if (weekNumber && year) {
      return year +', week '+ weekNumber;
    } else {
      return null;
    }
  }.property('year', 'weekNumber')
});
```
That way you could make it say anything you want, you could even depend on more than just the year and weekNumber (monthNumber maybe?).

Another option would be to instead make something like a `weekFormat` setting that takes a format string like `${year}-V${weekNumber}` or some similar format and then does a simple substring replace on `${year}` and `${weekNumber}`. Perhaps this is overkill though, given the first option.

I'm curious to hear what your thoughts are on this.

Also, on another note, there seems to be a failing test on travis. It does not seem like that should be caused by any of your changes though. I'll look into that over the weekend.

vasilakisfil commented:
About the tests, I gave it a quick try and it turns out that moment doesn't return the correct year when you are on the first week of the year (http://en.wikipedia.org/wiki/ISO_week_date#First_week) (we need an isoYear() method)

About the weekLetter, yeah it's not very helpful I agree. We had extended the component as you said but we thought we could make it as an injectable option. I am wondering, will it be able to inject a controller's method with htmlbars? That would be fun.

Ok I updated the pull-request to include only weekColumnHeader and fixed the tests.

RSSchermer commented on addon/components/rl-week-picker.js:
```diff
@@ -217,9 +219,11 @@ export default Ember.Component.extend(DropdownComponentMixin, {
   actions: {
     decreaseWeek: function () {
       if (!this.get('decreaseWeekButtonDisabled')) {
+        Ember.Logger.debug(this.currentMoment().format("LLL"));
```
Lingering debug line.

RSSchermer commented on addon/components/rl-week-picker.js:
```diff
@@ -217,9 +219,11 @@ export default Ember.Component.extend(DropdownComponentMixin, {
     decreaseWeek: function () {
       if (!this.get('decreaseWeekButtonDisabled')) {
+        Ember.Logger.debug(this.currentMoment().format("LLL"));
         var m = this.currentMoment().subtract(1, 'weeks');
```
I think maybe changing this line to:

```js
var m = this.currentMoment().isoWeekDay(4).subtract(1, 'weeks');
```
May solve the problem as well and then `if (newWeek === 52) { newYear -= 1; }` might not be necessary.

It's a really funny bug actually, because it seems like it depends on which day of the week you run the tests. If you create a moment from the week and year, it seems to set the weekday to the current weekday.

RSSchermer commented on README.md:
```diff
@@ -68,6 +68,9 @@ The following properties can be set to customize the month picker:
   `"<i class='fa fa-chevron-left'></i>"` to work with Font Awesome.
 * `nextPageButtonText` (default: '>'): : the text on the next page button. Set for example to
   `"<i class='fa fa-chevron-right'></i>"` to work with Font Awesome.
+* `weekLetter` (default: W): the week letter when formating the following week as text. Default is W so it is formated
```
Documentation remnant for the weekLetter setting.

RSSchermer commented:
I made a couple of inline comments. There's still a description of the weekLetter setting in the readme and there's a lingering debug line.

Also think that possibly setting the weekday to 4 may fix things as well (see inline comment) although I'm not 100% sure. I'd also be happy to merge this and play around with it myself a bit and see whats going on here exactly.

vasilakisfil commented:
I think it's ok now! Let me know if you need anything else!

RSSchermer commented:
Great! Thanks a lot for all the work! I'll release it as a new minor version.
